src/forms/FormControl.scss

```diff
@@ -104,166 +138,197 @@
+    position: absolute;
+    top: var(--base-size-4, 4px);
+    right: var(--base-size-4, 4px);
+    z-index: 4;
+    display: grid;
+    width: var(--primer-control-xsmall-size, 24px);
+    height: var(--primer-control-xsmall-size, 24px);
+    padding: 0;
+    color: var(--color-fg-muted);
+    cursor: pointer;
+    background: transparent;
+    border: 0;
+    border-radius: var(--primer-borderRadius-small);
+    transition: 0.2s cubic-bezier(0.3, 0, 0.5, 1);
+    transition-property: color, background-color, border-color;
+    align-items: center;
+    justify-content: center;
 
-  &.FormControl-fieldWrap--disabled {
-    color: var(--color-primer-fg-disabled);
-    background-color: var(--color-input-disabled-bg);
-  }
+    svg {
+      user-select: none;
+    }
 
-  &.FormControl-fieldWrap--invalid:not(:focus-within) {
-    box-shadow: var(--primer-borderInset-thin, inset 0 0 0 max(1px, 0.0625rem)) var(--color-danger-emphasis);
-  }
+    &[disabled] {
+      color: var(--color-primer-fg-disabled);
+      pointer-events: none;
+    }
 
-  // if leadingVisual is present
-  &.FormControl-fieldWrap--input-leadingVisual {
-    grid-template-columns: var(--base-size-16, 16px) minmax(0, auto);
+    &:hover {
+      background: var(--color-action-list-item-default-hover-bg);
+    }
 
-    .FormControl--input {
-      padding-inline-start: calc(var(--base-size-16, 16px) + var(--primer-control-medium-paddingInline-condensed, 8px));
+    &:active {
+      background: var(--color-action-list-item-default-active-bg);
     }
 
-    // if leadingVisual and trailingAction are present
-    &.FormControl-fieldWrap--input-trailingAction {
-      grid-template-columns: var(--base-size-16, 16px) minmax(0, auto) min-content;
+    // show vertical divider line between field and button
+    &::before {
+      position: absolute;
+      top: calc((var(--primer-control-xsmall-size) - var(--base-size-16)) / 2);
+      left: calc(var(--base-size-4, 4px) * -1);
+      display: block;
+      width: var(--primer-borderWidth-thin);
+      height: var(--base-size-16);
+      content: '';
+      background: var(--color-border-default);
+    }
 
-      .FormControl--input-trailingAction {
-        grid-column: 3 / 4;
+    &::after {
+      @include minTouchTarget(var(--primer-control-medium-size), var(--primer-control-medium-size));
+
+      @media (pointer: coarse) {
+        min-width: var(--primer-control-minTarget-coarse);
+        min-height: var(--primer-control-minTarget-coarse);
       }
     }
   }
 
-  // if trailingAction is present
-  &.FormControl-fieldWrap--input-trailingAction {
-    grid-template-columns: minmax(0, auto) min-content;
-  }
-
-  &.FormControl-fieldWrap--input {
-    grid-template-rows: max-content;
-
-    .FormControl--input-leadingVisual {
-      grid-column: 1 / 2;
-      grid-row: 1;
+  // if leadingVisual is present
+  &.FormControl-fieldWrap--input-leadingVisual {
+    .FormControl--input {
+      padding-inline-start:
```

Comment:
Ohh logical properties have entered the stylesheet! 🎉

Reply:
I need "logical properties have entered the stylesheet" on a shirt